www/src/utils/sidebar/item-list.js

```diff
@@ -1,67 +1,107 @@
+import { useContext } from "react"
+import { LocaleContext } from "../../components/layout"
+import { keyBy } from "lodash-es"
 import docsSidebar from "../../data/sidebars/doc-links.yaml"
 import contributingSidebar from "../../data/sidebars/contributing-links.yaml"
 import featuresSidebar from "../../data/sidebars/features-links.yaml"
 import tutorialSidebar from "../../data/sidebars/tutorial-links.yaml"
 
+import { useStaticQuery, graphql } from "gatsby"
+
 import { getLocaleAndBasePath } from "../i18n"
 
+function usePages() {
+  const { allMdx } = useStaticQuery(graphql`
+    query {
+      allMdx(limit: 10000, filter: { fileAbsolutePath: { ne: null } }) {
```

Comment:
This seems dangerous (as in it can bloat bundle). It also doesn't do anything to select only MDX nodes that it needs to (those that are actually linked from sidebars).

I don't have idea how to make it better yet, but will think about it. Probably will have to add some boolean flag to mdx nodes (something like "isLinkedInSidebar") that we can programatically do in `gatsby-node.js` and then only select those here

Reply:
With https://github.com/gatsbyjs/gatsby/issues/21323 , this can just be `allDocPages`, which will skip blog posts and package readmes.

Reply:
The way I see it - right now we could:
- either have this temporary thing for `isLinkedInSidebar` boolean field to not fetch whole lot of data that would be discarded
- or try to get #21323 in (which seems to depend on yet another PR)

Reply:
You're right, we should wait for that big change to unblock ourselves here. But I think I have a better idea than `isLinkedInSidebar`.